sigmadsp/communication/tcpip1701.py

```diff
@@ -0,0 +1,186 @@
+"""This module communicates with SigmaStudio.
+
+It can receive read/write requests and return with read response packets.
+"""
+import logging
+import socket
+import socketserver
+from typing import Union
+
+from sigmadsp.communication.base import (
+    ReadRequest,
+    SafeloadRequest,
+    ThreadedTCPServer,
+    WriteRequest,
+)
+from sigmadsp.helper.conversion import (
+    bytes_to_int8,
+    bytes_to_int16,
+    int8_to_bytes,
+    int16_to_bytes,
+)
+
+# A logger for this module
+logger = logging.getLogger(__name__)
+
+
+class SigmaStudioRequestHandler(socketserver.BaseRequestHandler):
```

Comment:
This is borrowing a lot from the old ADAU145x implementation. Most functionality is identical. I suggest to make a shared `SigmaStudioRequestHandler` base class and inherit from it.

Reply:
it already has a lot of the common parts moved into the parent class, I might have missed a few and maybe other things can be parameterized to deal with the differences; I'll take another look when I have time

Reply:
I ended up doing a bit more...